prime-router/src/main/kotlin/history/azure/DatabaseDeliveryAccess.kt

```diff
@@ -141,4 +155,112 @@ class DatabaseDeliveryAccess(
+                .from(
+                    ACTION.join(REPORT_FILE).on(
+                        REPORT_FILE.ACTION_ID.eq(ACTION.ACTION_ID)
+                    )
+                )
+                .where(filter)
+                .groupBy(REPORT_FILE.REPORT_ID)
+
+            query.fetchInto(ReportId::class.java)
+        }
+
+        return db.transactReturning { txn ->
+            var filter = COVID_RESULT_METADATA.REPORT_ID.isNotNull()
+            if (since != null) {
+                filter = filter.and(COVID_RESULT_METADATA.CREATED_AT.ge(since.toLocalDateTime()))
+            }
+            if (until != null) {
+                filter = filter.and(COVID_RESULT_METADATA.CREATED_AT.lt(until.toLocalDateTime()))
+            }
+
+            val sortedColumns = sortColumns.map {
+                val column = when (it) {
+                    /* Decides sort column by enum */
+                    BulkFacilitySortColumn.DATE -> ITEM_LINEAGE.CREATED_AT
+                    BulkFacilitySortColumn.PROVIDER -> COVID_RESULT_METADATA.ORDERING_PROVIDER_NAME
+                    BulkFacilitySortColumn.FACILITY -> COVID_RESULT_METADATA.TESTING_LAB_NAME
+                    BulkFacilitySortColumn.SUBMITTER -> COVID_RESULT_METADATA.SENDER_ID
+                    BulkFacilitySortColumn.REPORT_ID -> COVID_RESULT_METADATA.REPORT_ID
+                }
+
+                when (sortDir) {
+                    /* Applies sort order by enum */
+                    SortDir.ASC -> column.asc()
+                    SortDir.DESC -> column.desc()
+                }
+            }
+            val positive: Field<String> = DSL.field(
+                "sum(CASE WHEN test_result = 'DETECTED' THEN 1 ELSE 0 END)",
+                String::class.java
+            ).`as`("positive")
+            val total: Field<String> = DSL.field(
+                "count(covid_results_metadata_id)",
+                String::class.java
+            ).`as`("count_records")
+
+            val query = DSL.using(txn).select(
+                COVID_RESULT_METADATA.TESTING_LAB_NAME,
+                COVID_RESULT_METADATA.TESTING_LAB_CITY,
+                COVID_RESULT_METADATA.TESTING_LAB_STATE,
+                COVID_RESULT_METADATA.TESTING_LAB_CLIA,
+                positive,
+                total,
+                ITEM_LINEAGE.CREATED_AT,
+                COVID_RESULT_METADATA.ORDERING_PROVIDER_NAME,
+                COVID_RESULT_METADATA.SENDER_ID,
+                COVID_RESULT_METADATA.REPORT_ID,
+            ).from(
+                COVID_RESULT_METADATA.join(ITEM_LINEAGE).on(
+                    COVID_RESULT_METADATA.REPORT_ID.eq(ITEM_LINEAGE.ORIGIN_REPORT_ID),
+                    COVID_RESULT_METADATA.REPORT_INDEX.eq(ITEM_LINEAGE.ORIGIN_REPORT_INDEX)
+                )
+            ).where(
+                ITEM_LINEAGE.CHILD_REPORT_ID.`in`(reportIds)
+            ).and(filter)
+                .groupBy(
+                    COVID_RESULT_METADATA.TESTING_LAB_NAME,
```

Comment:
Looking at the mockup, do we want to group by provider, facility, submitter?

Reply:
We don't, that was a leftover from the calculated fields that are also being removed now